File: backend/internal/domain/department/department.go
package department

type Department struct {
	ID        int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	CompanyID int64  `gorm:"not null" json:"company_id"`
	ParentID  *int64 `gorm:"default:null" json:"parent_id"`
	Name      string `gorm:"type:varchar(100);not null" json:"name"`
	LeaderID  *int64 `json:"leader_id"`
	Status    int16  `gorm:"default:1" json:"status"`
}

func (Department) TableName() string {
	return "department"
}

// TreeNode 树形部门节点
type TreeNode struct {
	ID        int64       `json:"id"`
	CompanyID int64       `json:"company_id"`
	ParentID  *int64      `json:"parent_id"`
	Name      string      `json:"name"`
	LeaderID  *int64      `json:"leader_id"`
	Status    int16       `json:"status"`
	Children  []*TreeNode `json:"children,omitempty"`
}

File: frontend/src/api/department.ts
import { apiClient } from './client';

export interface Department {
  id: number
  name: string
  parent_id: number | null
  company_id: number
  leader_id?: number | null
  status: number
  children?: Department[]
}

export interface ApprovalChainStep {
  id?: number
  employee_id: number
  step_order: number
  employee_name?: string
}

export const departmentApi = {
  list: (companyId?: number) =>
    apiClient.get<Department[]>('/departments', {
      params: companyId ? { company_id: companyId } : {},
    }),

  create: (data: Partial<Department>) =>
    apiClient.post<Department>('/departments', data),

  update: (id: number, data: Partial<Department>) =>
    apiClient.put<Department>(`/departments/${id}`, data),

  delete: (id: number, transferToDeptId?: number) =>
    apiClient.delete(`/departments/${id}`, transferToDeptId ? { data: { transfer_to_dept_id: transferToDeptId } } : {}),

  getApprovalChain: (id: number) =>
    apiClient.get<ApprovalChainStep[]>(`/departments/${id}/approval-chain`),

  setApprovalChain: (id: number, steps: ApprovalChainStep[]) =>
    apiClient.put(`/departments/${id}/approval-chain`, { steps }),
};
